### PR TITLE
Bioprinter can print biotic limbs

### PR DESCRIPTION
**Pull request by Waselon**
## About The Pull Request

Bioprinter now can print biotic limbs which function like your regular organic limbs but take 130% damage as a downside. Those can be attached instead of robotic limbs.

No you cant attach delimbed organic limb back due to balance concerns.

Bioprinter comes preloaded with 1000 biomass
Printing one limb takes 100 biomass
One biomass beaker (60u) refills 400

You can recycle organic limbs for 50 biomass

## Why It's Good For The Game

Delimbs suck, but having a robotic prothesis as a human sucks even more.

## Changelog
:cl:
add: bioprinter can print biotic limbs, biotic limbs can be attached via surgery
/:cl:

### CODE REVIEW

**Comment by ZahaSanko** [comment]
at long last

File: code/game/objects/machinery/bioprinter.dm [hunk 1]
```diff
@@ -17,10 +17,14 @@
 
 	//"Name" = list(location, matter, metal, time, isorganic)
 	var/list/products = list(
-		"synthetic left arm (125 - Metal)" =  list(/obj/item/robot_parts/l_arm,  0, LIMB_METAL_AMOUNT, LIMB_PRINTING_TIME),
-		"synthetic right arm (125 - Metal)" = list(/obj/item/robot_parts/r_arm,  0, LIMB_METAL_AMOUNT, LIMB_PRINTING_TIME),
-		"synthetic left leg (125 - Metal)" =  list(/obj/item/robot_parts/l_leg,  0, LIMB_METAL_AMOUNT, LIMB_PRINTING_TIME),
-		"synthetic right leg (125 - Metal)" = list(/obj/item/robot_parts/r_leg,  0, LIMB_METAL_AMOUNT, LIMB_PRINTING_TIME)
+		"biotic left arm (125 - Metal)" =  list(/obj/item/limb/l_arm,  0, LIMB_METAL_AMOUNT, LIMB_PRINTING_TIME),
```
Comment (QualityVan) [inline code/game/objects/machinery/bioprinter.dm:20]: Made out of metal?

**Comment by Barnet2** [comment]
Holy fuck speedmerge this shit. Inshallah god bless.

File: code/modules/surgery/robolimbs.dm [hunk 2]
```diff
@@ -164,3 +164,49 @@
 	span_warning("Your hand slips, damaging connectors on [target]'s [affected.display_name]!"))
 	target.balloon_alert_to_viewers("Slipped!")
 	target.apply_damage(10, BRUTE, affected, 0, TRUE, updating_health = TRUE)
+
+/datum/surgery_step/limb/biotic/attach
```
Comment (QualityVan) [inline code/modules/surgery/robolimbs.dm:168]: Just change limb/attach to accept extra tools and do a check during end_step for whether to robotize or not, everything else here is duplicate code.

File: code/game/objects/machinery/bioprinter.dm (hunk 1 above)
Comment (Waselon) [reply code/game/objects/machinery/bioprinter.dm:20]: whatever

File: code/modules/surgery/robolimbs.dm (hunk 2 above)
Comment (Waselon) [reply code/modules/surgery/robolimbs.dm:168]: except you cant change var type and robot_perts is not a child of limb

**Comment by bowlofcereal** [comment]
image

behold, future vision

**Comment by Tupinambis** [comment]
Very good PR, delimbs shouldn't permanently screw you over, and this should make them less frustrating to deal with and make balancing them a bit easier.

**Comment by Derrick2Electricboogaloo** [comment]
So can we also reattach organic limbs recovered?

**Comment by QualityVan** [comment]
Yup. This PR would mean that if you get delimbed and someone has surgery tools nearby you can get fixed in about ten seconds with no resources spent.

**Comment by Derrick2Electricboogaloo** [comment]
> Yup. This PR would mean that if you get delimbed and someone has surgery tools nearby you can get fixed in about ten seconds with no resources spent.

:O The future is now

**Comment by MereWalker** [comment]
I'm still going to do robot arms

**Comment by Xander3359** [comment]
Isnt the whole point of delimbs that robot arms are designed to be unwanted? This is just blatantly powercreep. I agree with letting recovered limbs being re-attachable, but printing organic arms out of metal is ridiculous.

**Comment by Waselon** [comment]
> Isnt the whole point of delimbs that robot arms are designed to be unwanted? This is just blatantly powercreep. I agree with letting recovered limbs being re-attachable, but printing organic arms out of metal is ridiculous.

Theres no way to avoid delimbs, how can they be "unwanted"?

**Comment by Lumipharon** [comment]
I'm saying no due to various powergamey/memey stuff you can do with reattaching severed limbs.

If you make it only work with some specific 'vat grown limbs' or something I'll consider it, but tbh people should just deal with their robo limb, it's literally the only permenant thing in the game, and it's not even that bad considering the literal 50% damage resistance you get.

**Comment by Al-1ce** [comment]
> If you make it only work with some specific 'vat grown limbs' or something I'll consider it, but tbh people should just deal with their robo limb, it's literally the only permenant thing in the game, and it's not even that bad considering the literal 50% damage resistance you get.



It's a pain in the ass. And not in the "emergent gameplay" way we like to think of poorly designed mechanics. Nobody wants to bring engineering tools with them and deal with the slow ass repair because they got hurt.

If you want a metal limb, ask. Simply give marines a choice and you'll see that nobody wants one of these. The 50% DR is not worth the hassle.

File: code/modules/organs/limbs.dm [hunk 3]
```diff
@@ -860,6 +863,13 @@ Note that amputating the affected organ does in fact remove the infection from t
 		var/datum/limb/child_limb = c
 		child_limb.robotize()
 
+/datum/limb/proc/biotize()
```
Comment (Lumipharon) [inline code/modules/organs/limbs.dm:866]: autodoc

**Review by Lumipharon — CHANGES_REQUESTED** [review]

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**